### ISSUE TITLE
How to fetch all rows from a row-limited Portal?

### ISSUE THREAD

**Issue by fbernier** [sfackler/rust-postgres#552]
The documentation is not very clear around `query_portal` and `query_portal_raw`. What happens when you call it and there are no rows left? Does it return an Error or an empty vec / empty Iterator?

I think that fetching all rows of a query, but in batches, is the main use case of this and having an example in the documentation on how to do that with both would be helpful.

Thanks!

**Comment by sfackler** [comment]
If there are no rows left, it returns no rows.

**Comment by fbernier** [comment]
Sorry for the late reply. So basically the process to fetch all rows in batches when using `query_portal_raw` would be something like (untested. Will try it out soon):

```rust
        loop {
            let rows_iter = transaction.query_portal_raw(/*...*/);
            let has_rows = iter.peek().unwrap_or(None).is_some();
            if has_rows { /* consume row_iter */ } else { break }
        }
```

Is that it?

**Comment by sfackler** [comment]
That seems like it would work.

**Comment by nvanbenschoten** [comment]
@fbernier I don't know whether you're still interested in this, but I was also looking to do something similar and found that the behavior is more subtle than what's discussed here. It appears that `query_portal` (and presumably `query_portal_raw`) will reset the portal after any batch contains less than the desired `max_row` count. 

For instance, when fetching a set of 22 rows using a batch size of 10, I found that logic like what's posted above resulted in an infinite loop of:
```
row len 10
row len 10
row len 2
row len 10
row len 10
row len 2
...
```

Notice that the size of a batch never dropped down to 0, so the loop termination condition was never reached. Meanwhile, when fetching a set of 20 rows using a batch size of 10, I saw the expected:
```
row len 10
row len 10
row len 0
```

I don't know whether this is intentional or not - I'll defer to @sfackler on that. Either way, I ended up needing to write something that looked more like:
```rust
loop {
    let rows = transaction.query_portal(/*...*/, max_rows);
    let exhausted = rows.len() < max_rows;
    /* consume row_iter */
    if exhausted { break }
}
```
